packages/core/docs/guide/product-catalog.md

```diff
@@ -1,112 +1,206 @@
 # Product Catalog
 
-In Vue Storefront there are two composables available that you can use to interact with product catalog - `useProduct` and `useFacet`. The former should be used to work with a single product and its variants while the latter is for complex product listings with filtering and sorting.
+There are two composables used to interact with the product catalog - `useProduct` and `useFacet`.
 
+`useProduct` allows loading products and their variants when some parameters like IDs, SKUs, or slug are already known. 
 
-## Fetching a single product and it's variants with `useProduct`
+`useFacet` allows more complex queries where categories, search terms, filters, and other options can narrow down and sort the results.
 
-`useProduct` composable is meant to be used primarily on Product Details Page to display information about a single product and its variants.
+## Fetching a single product and its variants
 
-You can fetch the product and its variants with `search` method. The response will be saved in `products` object and contain at least an array of product variants.
+`useProduct` composable is used primarily on the Product Details Page to display information about a single product and its variants.
 
-```js
+Use the `search` method to fetch the product and its variants. The response is available in the `products` object.
+
+```vue
+<script>
 import { useProduct } from '{INTEGRATION}'
 import { onSSR } from '@vue-storefront/core'
-// ...
 
 export default {
   setup () {
-    const { products, search } = useProduct()
+    const { products, search } = useProduct();
 
     onSSR(async () => {
-      await search(searchParams) // populates 'products' with the result
+      /**
+       * "searchParams" may vary depending on the integration used.
+       * Please refer to its documentation for more details.
+       */
+      await search(searchParams);
     })
     
     return {
       products
-    }
+    };
   }
 }
+</script>
 
 ```
-### Using `getFiltered` to extract product list
 
-In some platforms `products` object is containing not only a list of products but also some additional metadata like products count. In such cases you can use `getFiltered` getter without parameters to extract only the products list.
+## Accessing product data
+
+Once products are loaded using `useProduct`, access them using `productGetters`. Depending on the product, configuration, and integration used, the response might contain one or more products or variants. Use the `getFiltered` getter to access them.
+
+For a full list of available getters, please refer to the [ProductGetters ](../core/api-reference/core.productgetters) interface.
 
-```ts
+### Accessing products list
+
+Use the `getFiltered` getter without the second parameter to get the list of all products.
+
+```vue
+<script>
 import { productGetters } from '{INTEGRATION}'
 
 export default {
   setup () {
-    // ...
+    // Load products as shown in the first example above
```

Comment:
Maybe we should just add this code here so that reader does not have to scroll to different sections?
Also `first example above` - kinda risky if we add some section before this one :)

Reply:
Do you have any suggestions on how to fix it? I don't want to copy so many lines from the first example, especially with the comment about `searchParams`.

Reply:
how about...
```js{1-2,7-9,14}
import { computed } from '@vue/composition-api';
import { productGetters } from '{INTEGRATION}';

export default {
  setup () {
    const { products, search } = useProduct();
    const productList = compute(() => {
      return productGetters.getFiltered(products.value);
    }
    
    search(/*  product search params  */);

    return {
      productList
    };
  };
};
```

with `js{1-2,7-9,14}`.  It doesn't seem to be that long to me and might be more precise. I dunno...

Also, in example above `import { computed }` might be worth adding :)